tests/epyccel/recognised_functions/test_numpy_types.py

```diff
@@ -0,0 +1,2173 @@
+
+    epyccel_func = epyccel(get_complex128, language=language)
+
+    assert epyccel_func(bl) == get_complex128(bl)
+    assert epyccel_func(integer8) == get_complex128(integer8)
+    assert epyccel_func(integer16) == get_complex128(integer16)
+    assert epyccel_func(integer) == get_complex128(integer)
+    assert epyccel_func(integer32) == get_complex128(integer32)
+    # the if block should be removed after resolving (https://github.com/pyccel/pyccel/issues/735).
+    if sys.platform != 'win32':
+        assert epyccel_func(integer64) == get_complex128(integer64)
+    assert epyccel_func(fl) == get_complex128(fl)
+    assert epyccel_func(fl32) == get_complex128(fl32)
+    assert epyccel_func(fl64) == get_complex128(fl64)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Arrays not handled yet."),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python]
+        )
+    )
+)
+
+def test_numpy_complex128_array_like_2d(language):
+
+    @types('bool[:,:]')
+    @types('int[:,:]')
+    @types('int8[:,:]')
+    @types('int16[:,:]')
+    @types('int32[:,:]')
+    @types('int64[:,:]')
+    @types('float[:,:]')
+    @types('float32[:,:]')
+    @types('float64[:,:]')
+    def get_complex128(arr):
+        from numpy import complex128, shape
+        a = complex128(arr)
+        s = shape(a)
+        return len(s), s[0], s[1], a[0,0], a[0,1]
+
+    size = (2, 5)
+
+    bl = randint(0, 1, size=size, dtype= bool)
+
+    integer8 = randint(min_int8, max_int8, size=size, dtype=np.int8)
+    integer16 = randint(min_int16, max_int16, size=size, dtype=np.int16)
+    integer = randint(min_int, max_int, size=size, dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=size, dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=size, dtype=np.int64)
+
+    fl = uniform(min_float / 2, max_float / 2, size = size)
+    fl32 = uniform(min_float32 / 2, max_float32 / 2, size = size)
+    fl32 = np.float32(fl32)
+    fl64 = uniform(min_float64 / 2, max_float64 / 2, size = size)
+
+    epyccel_func = epyccel(get_complex128, language=language)
+
+    assert epyccel_func(bl) == get_complex128(bl)
+    assert epyccel_func(integer8) == get_complex128(integer8)
+    assert epyccel_func(integer16) == get_complex128(integer16)
+    assert epyccel_func(integer) == get_complex128(integer)
+    assert epyccel_func(integer32) == get_complex128(integer32)
+    # the if block should be removed after resolving (https://github.com/pyccel/pyccel/issues/735).
+    if sys.platform != 'win32':
+        assert epyccel_func(integer64) == get_complex128(integer64)
+    assert epyccel_func(fl) == get_complex128(fl)
+    assert epyccel_func(fl32) == get_complex128(fl32)
+    assert epyccel_func(fl64) == get_complex128(fl64)
```

Comment:
I like the idea of a separate file for test_numpy_types :+1:
Could we put the functions in a separate file? That way you can reduce the number of tests by grouping tests of similar types. E.g:
```
@pytest.mark.parametrize( 'language', (
        pytest.param("fortran", marks = [pytest.mark.fortran]),
        pytest.param("c", marks = [
            pytest.mark.skip(reason="Arrays not handled yet."),
            pytest.mark.c]
        ),
        pytest.param("python", marks = [
            pytest.mark.python]
        )
    )
)
@pytest.mark.parametrize( 'get_complex', [mod.get_complex128, mod.get_complex64])
def test_numpy_complex128_array_like_2d(language, get_complex):
```